virtualcluster/pkg/syncer/conversion/mutate.go

```diff
@@ -172,7 +173,10 @@ func PodMutateDefault(vPod *v1.Pod, saSecretMap map[string]string, services []*v
 		if err != nil {
 			return err
 		}
-		mutateDNSConfig(p, vPod, vc.Spec.ClusterDomain, nameServer)
+		allowDNSPolicyMutation := featuregate.DefaultFeatureGate.Enabled(featuregate.TenantAllowDNSPolicyMutation)
+		if _, set := vPod.GetLabels()[constants.TenantDisableDNSPolicyMutation]; !allowDNSPolicyMutation && !set {
```

Comment:
The naming looks confusion. We call `mutateDNSConfig` when `allowDNSPolicyMutation` is false?

Reply:
good point, let me change that.